extensions/amp-animation/0.1/runners/native-web-animation-runner.js

```diff
@@ -208,11 +239,31 @@ export class NativeWebAnimationRunner extends AnimationRunner {
     });
   }
 
+  /**
+   * @override
+   */
+  finishOrPause() {
+    if (!this.isInitialized_()) {
+      return;
+    }
+    const players = this.players_;
+    this.players_ = null;
+    this.setPlayState_(WebAnimationPlayState.FINISHED);
+    players.forEach((player) => {
+      try {
+        // Will fail if animation is infinite, in that case we pause it.
+        player.finish();
+      } catch (error) {
+        player.pause();
+      }
+    });
```

Comment:
It's best if we don't duplicate implementations. Instead:

1. Reset `this.players_ = null` after `finish`ing all players.
2. Catch the `finish()` here. 

```suggestion
    try {
      this.finish();
    } catch {
      this.players_.forEach((player) => { player.pause() });
      this.players_ = null;
    }
```

Reply:
That implementation is missing the fact that a list of players can contain players that are infinite (and fail) and players that run a given amount of times (and succeed), so each player could fail or succeed to `finish()` and catching the call to `.finish()` does not tell us what players succeeded or failed, and which ones need to be paused. Moreover, let's say the first player is infinite, it will not call `finish` on all the remaining ones because the code threw an error.

Reply:
Makes sense. Let's make this the implementation for `finish()`, but pause only when provided with a `pauseOnError` argument, which would be false by default.